t/t-push.sh

```diff
@@ -740,3 +740,34 @@ begin_test 'push with data the server already has'
   assert_server_object "$reponame" "$contents2_oid"
 )
 end_test
+
+begin_test "push custom reference"
+(
+  set -e
+
+  reponame="push-custom-reference"
+  setup_remote_repo "$reponame"
+  clone_repo "$reponame" "$reponame"
+
+  git lfs track "*.dat"
+  # generate content we'll use
+  content="filecontent"
+  oid=$(calc_oid "$content")
+
+  echo "[
+  {
+    \"CommitDate\":\"$(get_date -6m)\",
+    \"Files\":[
+      {\"Filename\":\"file.dat\",\"Size\":${#content}, \"Data\":\"$content\"}]
+  }
+  ]" | lfstest-testutils addcommits
+
+  # Create and try pushing a reference in a nonstandard namespace, that is,
+  # outside of refs/heads, refs/tags, and refs/remotes.
+  mkdir -p .git/refs/custom/remote/heads
+  cp .git/refs/heads/master .git/refs/custom/remote/heads/
```

Comment:
Can we write this as `git update-ref refs/custom/remote/heads/master refs/heads/master`?  That's generally the preferred way of creating arbitrary refs in Git, and we do that in some other places in the testsuite.

Reply:
Also, the `mkdir` won't be necessary then, since Git will create it for us.

Reply:
@bk2204: Very good suggestion, thanks for pointing this out. I changed the test as per your suggestion. I didn’t know about `git update-ref` in fact, so thanks for teaching me this one!